xy/point_centroid.go

```diff
@@ -16,6 +16,20 @@ func PointsCentroid(point *geom.Point, extra ...*geom.Point) geom.Coord {
 	return calc.GetCentroid()
 }
 
+// MultiPointCentroid computes the centroid of the multi point argument
+//
+// Algorithm: average of all points
+func MultiPointCentroid(point *geom.MultiPoint) geom.Coord {
+	calc := NewPointCentroidCalculator()
+	coords := point.FlatCoords()
+	stride := point.Layout().Stride()
+	for i := 0; i < len(coords); i += stride {
```

Comment:
Hmmm, not sure about this one now. On one side I think it makes sense to use the external API exported by go-geom (i.e. `NumPoints()` and `Point()`) for clarity, on the other side, this is perfectly reasonable use of flat coordinates for performance (avoiding a function call and allocation).

So, I retract my suggestion :)